### DISCUSSION TITLE
CR5 Pro Source firmware

### DISCUSSION THREAD

**Discussion by Randwic** [CrealityOfficial/Creality_Sonic_Pad#18] [General]
Hello,

I would like to get the source firmware for the CR5 PRO, but it is not available anywhere.

I would like to get it because I have a big problem with the platter heating up when I am at the end of the coil.

Indeed when the end of the filament is detected, the heating of the plate is cut, and if I'm not in front of my printer, when I resume the print is taken off the plate. So I don't see the use of the filament sensor.

Thanks in advance

**Comment by kamoteshake** [comment]
I don't have a CR 5 Pro but I support this. @CrealityTech Please release the source code. 🙏

**Comment by Randwic** [comment]
Thanks,Would it be possible to let me know when this one will be realease?

**Reply by kamoteshake** [reply]
I always keep an eye in the Creality repos. I've been waiting for them to release the source code of the latest Ender 3 S1 Plus firmware. so yeah. If I see something, I'll ping you here.

**Reply by Randwic** [reply]
Ok but considering their reactivity on the messages I have the impression that it won't happen right away.

**Reply by kamoteshake** [reply]
Well based on the Ender 3 S1 Pro, People started asking for it on Nov 2022 and they released it around Jan 2023. so around 2 months. 😅I guess it's not that bad. but yes. it seems like they only release the firmware if someone asks for it.

**Reply by Randwic** [reply]
I would just like to start:

-Know the environment
- to know which drivers are installed

and especially to know how to leave the bed in heating when I am at the end of filament.

I hope they will be quick to answer

**Reply by kamoteshake** [reply]
What board comes with your CR 5 Pro? Maybe we could search the drivers that comes with it. for the filament runout sensor. a quick search, it seems like we have to configure the firmware. I'll keep looking to see if there are any gcode somewhere that we could use.

**Reply by kamoteshake** [reply]
So I did more searching. it seems like the bed temp is fixed in the latest Marlin firmware. make sure you update your firmware. looks like CR 5 Pro comes with the 4.2.2 board. It seems like that board comes with either hr4988/a4988 or tmc2208

**Reply by Randwic** [reply]
Thank you very much for this information, now we just have to wait for the source FW.

**Comment by matsura8** [comment]
To add to the request, my new CR-5 is bottoming out the nozzle to the bed when it auto-homes. The z-offset does not seem to move the platform permanently. Apparently there is a Z-offset wizard in Marlin firmware to calibrate, but I need the source firmware.
